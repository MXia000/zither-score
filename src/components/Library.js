import React, { Component } from 'react';
import NavBar from './NavBar';
import { Auth } from 'aws-amplify';

class Library extends Component {
    render () {
        return (
            <div>
                    <div id="nav-bar">
                        <NavBar></NavBar>
                    </div>
            </div>
        )
    }
}

export default Library;