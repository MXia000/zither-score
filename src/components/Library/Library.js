import React, { Component } from 'react';
import CreateModal from "../CreateModal/CreateModal";
import { Navbar, Nav, NavDropdown, Form, FormControl, Button } from "react-bootstrap";
import { Auth, graphqlOperation, API } from 'aws-amplify';
import * as mutations from '../../graphql/mutations';

class Library extends Component {
    constructor(props) {
        super(props);
        this.state = {
          modal: false
        }
        this.handleShow = this.handleShow.bind(this);
    }
    
    handleShow = () => {
    this.setState(prevState => {
        return {
        modal: !prevState.modal
        };
    });
    };

    render () {
        return (
            <div className="main-library">
                <div className="side-bar">
                    <div className="list-tree">
                        <div className="inner">
                                <a className="list-item">
                                    <span className="t">Scores</span>
                                </a>
                                <div className="sub-item">
                                    <a className="list-item">
                                        <span className="t em">New collection</span>
                                    </a>
                                </div>
                                <a className="list-item">
                                    <span className="t">Trash</span>
                                </a>
                        </div>
                    </div>
                </div>
            
                <div className="main-bar">
                    <div className="library-content">
                        <div className="head-container">
                            <div className="header">
                                <h2>
                                    Scores
                                </h2>
                            </div>
                        </div>
                        
                        <div className="list-container">
                            <div className="library-list">
                                <div className="thead">
                                    <div className="tr">
                                        {/* <div className="th cb">
                                            <input type="checkbox" className="checkbox-morph"></input>
                                        </div> */}
                                        <div className="th row-title sorting">
                                            Name
                                        </div>
                                        <div className="th row-date sorting">
                                            Date Modified
                                        </div>
                                        <div className="th row-sharing">
                                            Sharing
                                        </div>
                                    </div>
                                </div>

                                <div infinite-scroll-disabled="infiniteScrollBusy" infinite-scroll-distance="250" className="tbody">

                                </div>
                            </div>
                        </div>
                    </div>

                    <div className="list-actions">
                        <div className="inner">
                            <button onClick={this.handleShow} className="btn-lg btn-teal-gradient main-action">Create new score</button>
                        </div>
                    </div>
                </div>
                <CreateModal
                    modal = {this.state.modal}
                    handleShow = {this.handleShow}
                />
            </div>
        )
    }
}

export default Library;