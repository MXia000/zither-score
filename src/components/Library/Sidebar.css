.sidebar {
    height: 100%;
    width: 160px; /* Set the width of the sidebar */
    position: fixed; /* Fixed Sidebar (stay in place on scroll) */
    z-index: 1; /* Stay on top */
    top: 70px; /* Stay at the top */
    left: 0;
    box-shadow: 0 3px 5px 2px rgba(255, 105, 135, .3);
    color: white;
    overflow-x: hidden; /* Disable horizontal scroll */
    padding-top: 20px;
  }
  
  /* The navigation menu links */
  .sidebar a {
    padding: 6px 8px 6px 16px;
    text-decoration: none;
    font-size: 20px;
    color: black;
    display: block;
  }
  
  /* When you mouse over the navigation links, change their color */
  .sidebar a:hover {
    color: #f1f1f1;
  }
  
  /* Style page content */
  .main {
    margin-left: 160px; /* Same as the width of the sidebar */
    padding: 0px 10px;
    color: black;
  }
  
  /* On smaller screens, where height is less than 450px, change the style of the sidebar (less padding and a smaller font size) */
  @media screen and (max-height: 450px) {
    .sidebar {padding-top: 15px;}
    .sidebar a {font-size: 18px;}
  }