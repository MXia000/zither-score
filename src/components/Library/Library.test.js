import React from 'react';
import { shallow } from 'enzyme';
import Library from './Library';

describe("Library", () => {
    it('render library without crashing', () => {  
        const LibraryComp = shallow(<div><Library /></div>);
        expect(LibraryComp).toMatchSnapshot();
    });
});