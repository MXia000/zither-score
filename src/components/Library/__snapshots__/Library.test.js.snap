// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library render library without crashing 1`] = `ShallowWrapper {}`;
